database/src/main/java/database/controller/DatabaseServiceRestController.java

```diff
@@ -80,13 +80,35 @@ void handleGetByFilters(Context ctx) {
 
     void handleGetAllRecords(Context ctx) {
         Class<? extends BaseEntity> entityClass = getClassFromPath(ctx);
-        ctx.json(databaseService.getAllRecordsFromTable(entityClass));
+        String limitParam = ctx.queryParam("limit");
+        String offsetParam = ctx.queryParam("offset");
+
+        if (limitParam != null && offsetParam != null) {
+            try {
+                int limit = Integer.parseInt(limitParam);
+                int offset = Integer.parseInt(offsetParam);
+
+                ctx.json(databaseService.getAllRecordsFromTable(entityClass, limit, offset));
+            } catch (NumberFormatException e) {
+                ctx.status(400).result("Invalid value for limit or offset parameter. They must be integers.");
```

Comment:
Please throw `BadRequestException` instead. We want to return all errors in one format. See how it works in `FileDatabaseExceptionHandler`.

Reply:
Btw, please rename `FileDatabaseExceptionHandler` class to `DatabaseControllerExceptionHandler` to make it more common.